apps/meteor/app/lib/server/functions/deleteRoom.ts

```diff
@@ -2,16 +2,24 @@ import { Messages, Rooms, Subscriptions } from '@rocket.chat/models';
 
 import { callbacks } from '../../../../lib/callbacks';
 import { FileUpload } from '../../../file-upload/server';
-import { notifyOnRoomChangedById } from '../lib/notifyListener';
+import { notifyOnRoomChangedById, notifyOnSubscriptionChanged } from '../lib/notifyListener';
 
 export const deleteRoom = async function (rid: string): Promise<void> {
 	await FileUpload.removeFilesByRoomId(rid);
+
 	await Messages.removeByRoomId(rid);
+
 	await callbacks.run('beforeDeleteRoom', rid);
-	await Subscriptions.removeByRoomId(rid);
+
+	await Subscriptions.removeByRoomId(rid, {
+		async onTrash(doc) {
+			void notifyOnSubscriptionChanged(doc, 'removed');
+		},
+	});
+
 	await FileUpload.getStore('Avatars').deleteByRoomId(rid);
+
 	await callbacks.run('afterDeleteRoom', rid);
-	await Rooms.removeById(rid);
 
-	void notifyOnRoomChangedById(rid, 'removed');
+	(await Rooms.removeById(rid)).deletedCount && void notifyOnRoomChangedById(rid, 'removed');
```

Comment:
Other places are using the `if (deletedCount)`, think we should keep the same standard here.